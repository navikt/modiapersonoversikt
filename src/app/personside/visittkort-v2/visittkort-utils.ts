import { ENDASH, formaterDato } from '../../../utils/string-utils';
import {
    AdresseBeskyttelse,
    ForelderBarnRelasjon,
    ForelderBarnRelasjonRolle,
    KodeBeskrivelse,
    Navn,
    Person,
    Sivilstand,
    SivilstandType
} from './PersondataDomain';

function erDod(person: Person) {
    return !person.dodsdato.isEmpty();
}

export function hentBarnUnder22(forelderBarnRelasjon: ForelderBarnRelasjon[]) {
    return hentBarn(forelderBarnRelasjon).filter((barn) => hentAlderOrDefault(barn) <= 21);
}

export function hentBarn(forelderBarnRelasjon: ForelderBarnRelasjon[]) {
    return forelderBarnRelasjon.filter((relasjon) => relasjon.rolle === ForelderBarnRelasjonRolle.BARN);
}

function hentAlderOrDefault(relasjon: ForelderBarnRelasjon) {
    return relasjon.alder ? relasjon.alder : 0;
}

export function hentForeldre(relasjoner: ForelderBarnRelasjon[]) {
    return relasjoner.filter(
        (relasjon) =>
            relasjon.rolle === ForelderBarnRelasjonRolle.FAR ||
            relasjon.rolle === ForelderBarnRelasjonRolle.MOR ||
            relasjon.rolle === ForelderBarnRelasjonRolle.MEDMOR
    );
}

export function harDiskresjonskode(adressebeskyttelse: KodeBeskrivelse<AdresseBeskyttelse>[]) {
    return adressebeskyttelse.some(
        (beskyttelse) =>
            beskyttelse.kode === AdresseBeskyttelse.KODE6 ||
            beskyttelse.kode === AdresseBeskyttelse.KODE6_UTLAND ||
            beskyttelse.kode === AdresseBeskyttelse.KODE7
    );
}

export function erPartner(sivilstand: Sivilstand): boolean {
    const aktuelleRelasjoner = [SivilstandType.GIFT, SivilstandType.REGISTRERT_PARTNER];
    return sivilstand?.type && aktuelleRelasjoner.includes(sivilstand?.type.kode);
}

export function hentNavn(navn: Navn | null, feilmelding: string = 'Ukjent navn'): string {
    if (!navn) {
        return feilmelding;
    }
    return navn.fornavn + (navn.mellomnavn ? ' ' + navn.mellomnavn + ' ' : ' ') + navn.etternavn;
}

export function hentAlderEllerDod(person: Person): string | undefined {
    if (erDod(person)) {
        return 'Død';
    }
    return person.alder?.toString();
}

export function hentAlderEllerDodRelasjon(relasjon: ForelderBarnRelasjon): string | undefined {
    if (relasjonErDod(relasjon)) {
        return 'Død';
    }
    return relasjon.alder?.toString();
}

export function relasjonErDod(relasjon: ForelderBarnRelasjon): boolean {
    return !relasjon.dodsdato.isEmpty();
}

export function hentPeriodeTekst(gyldighetstidspunkt: string | null, opphorstidspunkt?: string | null) {
    const fom = gyldighetstidspunkt ? formaterDato(gyldighetstidspunkt) : '';
    const tom = opphorstidspunkt ? formaterDato(opphorstidspunkt) : '';

    return `${fom} ${ENDASH} ${tom}`;
}
