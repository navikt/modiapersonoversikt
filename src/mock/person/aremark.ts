import { Diskresjonskoder, TilrettelagtKommunikasjonsTyper } from '../../konstanter';
import { Kjønn, Person, Relasjonstype, SivilstandTyper } from '../../models/person/person';

export const aremark: Person = {
    fødselsnummer: '10108000398',
    kjønn: Kjønn.Mann,
    geografiskTilknytning: '0118',
    alder: 42,
    navn: {
        sammensatt: 'AREMARK TESTFAMILIEN',
        fornavn: 'AREMARK',
        mellomnavn: '',
        etternavn: 'TESTFAMILIEN',
    },
    tilrettelagtKomunikasjonsListe: [
        {
            behovKode: 'TOHJ',
            beskrivelse: TilrettelagtKommunikasjonsTyper.TOHJ
        }
    ],
    diskresjonskode: Diskresjonskoder.FORTROLIG_ADRESSE,
    statsborgerskap: 'NORSK',
    personstatus: {
        dødsdato: undefined,
        bostatus: undefined
    },
    sivilstand: {
        value: SivilstandTyper.Gift,
        beskrivelse: 'Gift',
        fraOgMed: '2005-12-12'
    },
    familierelasjoner: [{
        harSammeBosted: false,
        rolle: Relasjonstype.Barn,
        tilPerson: {
            navn: {
                sammensatt: '',
                fornavn: 'Siri',
                mellomnavn: null,
                etternavn: 'Aremark'
            },
            alder: 17,
            fødselsnummer: '10117100000',
            personstatus: {}
        }

    }],
    kontaktinformasjon: {
        mobil: {
            sistEndret: '2014-06-21T18:44:39+02:00',
            telefonnummer: '99887766',
            retningsnummer: '47',
            sistEndretAv: 'BRUKER'
        }
    }
};