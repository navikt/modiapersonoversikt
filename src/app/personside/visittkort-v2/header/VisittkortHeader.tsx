import * as React from 'react';
import { Data as Persondata, Kjonn, KodeBeskrivelse } from '../PersondataDomain';
import { Undertittel } from 'nav-frontend-typografi';
import styled from 'styled-components/macro';
import theme, { pxToRem } from '../../../../styles/personOversiktTheme';
import { hentAlderEllerDod, hentNavn } from '../visittkort-utils';
import { useRef } from 'react';
import PersonStatus from './status/PersonStatus';
import Etiketter from './etiketter/Etiketter';
import VisMerChevron from '../../../../components/VisMerChevron';
import NavKontorContainer from './navkontor/NavKontorContainer';
import { useOnMount } from '../../../../utils/customHooks';
import KjonnIkon from './KjonnIkon';

interface Props {
    persondata: Persondata;
    erApen: boolean;
    toggleApen: () => void;
}

const VisittkortHeaderDiv = styled.section`
    background-color: white;
    padding: ${theme.margin.layout};
    padding-right: 0;
    position: relative;
    display: flex;
    flex-wrap: nowrap;
    width: 100%;
    cursor: pointer;
`;

const StyledContent = styled.div`
    flex-grow: 1;
    display: flex;
    flex-flow: row wrap;
    > * {
        flex: 1 1 15rem;
    }
`;

const VenstreFelt = styled.section`
    display: flex;
    min-width: 35%;
`;

const HoyreFelt = styled.div`
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    text-align: right;
    box-sizing: border-box;
`;

const GrunninfoDiv = styled.section`
    flex: 1 1;
    text-align: left;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    word-break: break-word;
    *:focus {
        ${theme.focus}
    }
    > *:first-child {
        margin-bottom: 0.2em !important;
    }
`;

const ChevronStyling = styled.div`
    padding: ${pxToRem(10)};
    display: flex;
    align-items: center;
    justify-content: center;
`;

const ukjentKjonn: KodeBeskrivelse<Kjonn> = {
    kode: Kjonn.U,
    beskrivelse: 'Ukjent kjønn'
};

function VisittkortHeader(props: Props) {
    const navneLinjeRef = useRef<HTMLSpanElement>(null);
    const person = props.persondata.person;

    useOnMount(() => {
        if (!person.sikkerhetstiltak.isEmpty()) {
            return;
        }
        navneLinjeRef.current?.focus();
    });

    const kjonn = person.kjonn.firstOrNull() ?? ukjentKjonn;
    const navn = person.navn.firstOrNull();

    return (
        <VisittkortHeaderDiv role="region" aria-label="Visittkort-hode" onClick={props.toggleApen}>
            <StyledContent>
                <VenstreFelt>
                    <KjonnIkon kjonn={kjonn.kode} />
                    <GrunninfoDiv>
                        <Undertittel tag="h1">
                            <span ref={navneLinjeRef} tabIndex={-1} /* for at focus skal funke */>
                                {hentNavn(navn)} ({hentAlderEllerDod(person)})
                            </span>
                        </Undertittel>
                        <span className="visually-hidden">{kjonn.beskrivelse}</span>
                        <PersonStatus person={person} />
                    </GrunninfoDiv>
                </VenstreFelt>
                <HoyreFelt>
                    <Etiketter person={person} />
                    <NavKontorContainer person={person} />
                </HoyreFelt>
            </StyledContent>
            <ChevronStyling>
                <VisMerChevron
                    onClick={(e) => {
                        e.stopPropagation();
                        props.toggleApen();
                    }}
                    open={props.erApen}
                    title={(props.erApen ? 'Lukk' : 'Åpne') + ' visittkort (Alt + N)'}
                    focusOnRelativeParent={true}
                >
                    <span className="visually-hidden">{props.erApen ? 'Lukk visittkort' : 'Ekspander visittkort'}</span>
                </VisMerChevron>
            </ChevronStyling>
        </VisittkortHeaderDiv>
    );
}

export default VisittkortHeader;
