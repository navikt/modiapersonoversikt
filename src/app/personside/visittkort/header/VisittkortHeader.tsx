import * as React from 'react';
import styled from 'styled-components';
import Undertittel from 'nav-frontend-typografi/lib/undertittel';
import NavKontorContainer from './navkontor/NavKontorContainer';
import { erDød, Navn, Person } from '../../../../models/person/person';
import PersonStatus from './status/PersonStatus';
import EtiketterContainer from './Etiketter/EtiketterContainer';
import Mann from '../../../../svg/Mann.js';
import Kvinne from '../../../../svg/Kvinne.js';
import VisMerChevron from '../../../../components/VisMerChevron';
import theme from '../../../../styles/personOversiktTheme';
import ToolTip from '../../../../components/tooltip/ToolTip';
import IfFeatureToggleOn from '../../../../components/featureToggle/IfFeatureToggleOn';
import { FeatureToggles } from '../../../../components/featureToggle/toggleIDs';

interface Props {
    visittkortApent: boolean;
    person: Person;
    toggleVisittkort: (erApen?: boolean) => void;
}

interface State {
    showTooltip: boolean;
}

const VisittkortHeaderDiv = styled.section`
    ${theme.hvittPanel};
    padding: ${theme.margin.px20};
    padding-right: 3rem;
    margin-bottom: 0.2rem;
    position: relative;
    width: 100%;
    display: flex;
    flex-flow: row wrap;
    cursor: pointer;
    > * {
        flex: 1 1 50%;
    }
`;

const VenstreFelt = styled.section`
    display: flex;
`;

const HøyreFelt = styled.div`
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    text-align: right;
    box-sizing: border-box;
`;

const IkonDiv = styled.div`
    flex: 0 0 ${theme.margin.px50};
    text-align: left;
    svg {
        height: ${theme.margin.px40};
        width: auto;
    }
`;

const GrunninfoDiv = styled.section`
    flex: 1 1;
    text-align: left;
    display: flex;
    flex-flow: column nowrap;
    justify-content: space-between;
    *:focus {
        ${theme.focus}
    }
    > *:first-child {
        margin-bottom: 0.2em !important;
    }
`;

const ChevronStyling = styled.div`
    position: absolute;
    right: 1rem;
    top: 50%;
    transform: translateY(-50%);
`;

function hentNavn(navn: Navn) {
    return navn.fornavn + (navn.mellomnavn ? ' ' + navn.mellomnavn + ' ' : ' ') + navn.etternavn;
}

class VisittkortHeader extends React.PureComponent<Props, State> {
    private navneLinjeRef = React.createRef<HTMLSpanElement>();

    constructor(props: Props) {
        super(props);
        this.state = { showTooltip: false };
        this.toggleVisittkort = this.toggleVisittkort.bind(this);
    }

    componentDidMount() {
        if (this.navneLinjeRef.current && !this.props.person.sikkerhetstiltak) {
            this.navneLinjeRef.current.focus();
        }
    }

    toggleVisittkort() {
        this.setState({ showTooltip: true });
        this.props.toggleVisittkort(!this.props.visittkortApent);
    }

    render() {
        const { person, visittkortApent } = this.props;
        const ikon = {
            ikon: person.kjønn === 'M' ? <Mann /> : <Kvinne />
        };
        const alder = erDød(person.personstatus) ? 'Død' : person.alder;
        const kjønn = person.kjønn === 'M' ? 'Mann' : 'Kvinne';
        return (
            <VisittkortHeaderDiv role="region" aria-label="Visittkort-hode" onClick={this.toggleVisittkort}>
                <VenstreFelt>
                    <IkonDiv>{ikon.ikon}</IkonDiv>
                    <GrunninfoDiv>
                        <Undertittel tag="h1">
                            <span ref={this.navneLinjeRef} tabIndex={-1} /* for at focus skal funke*/>
                                {hentNavn(person.navn)} ({alder})<span className="visually-hidden">{kjønn}</span>
                            </span>
                        </Undertittel>
                        <PersonStatus person={person} />
                    </GrunninfoDiv>
                </VenstreFelt>

                <HøyreFelt>
                    <EtiketterContainer />
                    <NavKontorContainer person={person} />
                </HøyreFelt>

                <ChevronStyling>
                    <VisMerChevron
                        onClick={this.toggleVisittkort}
                        open={visittkortApent}
                        title={(visittkortApent ? 'Lukk' : 'Åpne') + ' visittkort (Alt + N)'}
                    >
                        <span className="visually-hidden">
                            {visittkortApent ? 'Lukk visittkort' : 'Ekspander visittkort'}
                        </span>
                    </VisMerChevron>
                </ChevronStyling>
                <IfFeatureToggleOn toggleID={FeatureToggles.Tooltip}>
                    {this.state.showTooltip && <ToolTip>Hurtigtast åpne/lukke visittkort: Alt + N</ToolTip>}
                </IfFeatureToggleOn>
            </VisittkortHeaderDiv>
        );
    }
}

export default VisittkortHeader;
