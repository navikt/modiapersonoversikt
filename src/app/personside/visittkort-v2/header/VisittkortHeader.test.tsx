import { act } from '@testing-library/react';
import { aremark } from '../../../../mock/persondata/aremark';
import { renderWithProviders } from '../../../../test/Testprovider';
import type { LocalDate, Person, Data as PersonData } from '../PersondataDomain';
import { hentNavn } from '../visittkort-utils';
import VisittkortHeader from './VisittkortHeader';

function lagPersondataForAremark(): PersonData {
    const person = aremark;
    return { feilendeSystemer: [], person };
}

describe('Tester visittkort-header sin funksjonalitet', () => {
    test('viser info om bruker i visittkort-header', async () => {
        const visittkortHeader = await act(() =>
            renderWithProviders(
                <VisittkortHeader persondata={lagPersondataForAremark()} erApen={false} toggleApen={() => null} />
            )
        );

        const json = visittkortHeader.asFragment();
        expect(json).toMatchSnapshot();
    });

    test('setter fokus p책 brukerens navn on mount', async () => {
        const person: Person = {
            ...aremark,
            sikkerhetstiltak: []
        };

        const oppdatertPerson: PersonData = { feilendeSystemer: [], person };
        await act(() =>
            renderWithProviders(
                <VisittkortHeader persondata={oppdatertPerson} erApen={false} toggleApen={() => null} />
            )
        );

        const focusedElement = document.activeElement;

        if (focusedElement) {
            expect(focusedElement).toHaveTextContent(new RegExp(`${hentNavn(person.navn[0])}`, 'i'));
        } else {
            assert.fail('Ingen activeElement p책 dokumentet');
        }
    });

    test('setter fokus p책 sikkerhetstiltak on mount', async () => {
        const person: Person = {
            ...aremark,
            sikkerhetstiltak: [
                {
                    type: 'TFUS',
                    beskrivelse: 'Telefonisk utestengelse',
                    gyldighetsPeriode: {
                        gyldigFraOgMed: '2020-01-01' as LocalDate,
                        gyldigTilOgMed: '2023-01-01' as LocalDate
                    }
                }
            ]
        };

        const oppdatertPerson: PersonData = { feilendeSystemer: [], person };

        await act(() =>
            renderWithProviders(
                <VisittkortHeader persondata={oppdatertPerson} erApen={false} toggleApen={() => null} />
            )
        );

        const focusedElement = document.activeElement;

        if (focusedElement) {
            expect(focusedElement).toHaveTextContent('Sikkerhetstiltak');
        } else {
            // eslint-disable-next-line @typescript-eslint/no-unsafe-call
            assert.fail('Ingen activeElement p책 dokumentet');
        }
    });
});
