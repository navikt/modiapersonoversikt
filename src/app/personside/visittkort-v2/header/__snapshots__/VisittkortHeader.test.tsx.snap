// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tester visittkort-header sin funksjonalitet > viser info om bruker i visittkort-header 1`] = `
.c7 {
  position: absolute;
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c6 {
  padding: 0 0 0 0.2rem;
  border: none;
  cursor: pointer;
  background-color: transparent;
  border-radius: 0.2rem;
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
}

.c6:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c6 svg {
  width: 1rem;
  color: #005b82;
  stroke-width: 1.4;
}

.c6:hover {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c6:active {
  opacity: 0.5;
}

.c6:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c10 {
  width: 1rem;
  margin-top: -0.125rem;
  margin-left: 0.125rem;
  margin-right: 0.125rem;
}

.c8 {
  margin: 0;
  padding: 0;
  list-style: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c8 li {
  white-space: nowrap;
}

.c8 li:not(:last-child):after {
  content: '/';
  margin: 0 0.2em;
}

.c5 {
  margin-right: 1em;
}

.c5 > *:not(:first-child)::before {
  content: '–';
  margin: 0 0.5rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c12 > * {
  margin: 6px 0 0 6px;
  white-space: nowrap;
}

.c15 {
  border: none;
  cursor: pointer;
  background-color: transparent;
  padding: 0.1rem;
  border-radius: 0.5em;
}

.c15:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c15:focus {
  outline: none;
  box-shadow: none;
}

.c15:focus::after {
  position: absolute;
  top: 0;
  left: 0;
  content: '';
  box-shadow: inset 0 0 0 0.1875rem #254b6d;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c13 {
  margin: 0.5rem 0 0 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  margin-right: 0.5rem;
}

.c3 svg {
  height: 2.5rem;
  width: auto;
}

.c0 {
  background-color: white;
  padding: .5rem;
  padding-right: 0;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  cursor: pointer;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c1 > * {
  -webkit-flex: 1 1 15rem;
  -ms-flex: 1 1 15rem;
  flex: 1 1 15rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-width: 35%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: right;
  box-sizing: border-box;
}

.c4 {
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  text-align: left;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  word-break: break-word;
}

.c4 *:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c4 > *:first-child {
  margin-bottom: 0.2em !important;
}

.c14 {
  padding: 0.625rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<section
  aria-label="Visittkort-hode"
  className="c0"
  onClick={[Function]}
  role="region"
>
  <div
    className="c1"
  >
    <section
      className="c2"
    >
      <div
        className="c3"
      >
        <svg
          alt="Mann"
          aria-label="Mann"
          height={64}
          viewBox="0 0 64 64"
          width={64}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 0a32 32 0 1 0 0 64 32 32 0 0 0 0-64zm0 8a3.9 3.9 0 1 1 0 7.7 3.9 3.9 0 0 1 0-7.8zm10 15.5v10.8c0 .9-.7 1.6-1.6 1.6-1 0-1.8-.7-1.8-1.6v-8.9c0-.4-.3-.8-.7-.8-.4 0-.7.3-.7.7v28.5a2.2 2.2 0 1 1-4.5 0v-18a.7.7 0 0 0-1.4 0v18a2.2 2.2 0 0 1-4.4 0V25.4a.7.7 0 0 0-1.5 0V34.3c0 .9-.8 1.6-1.8 1.6s-1.7-.7-1.7-1.6V23.5v-.7c0-3.4 3.6-5.8 7-6H35c3.4.2 7 2.6 7 6v.7z"
            fill="#3385D1"
          />
        </svg>
      </div>
      <section
        className="c4"
      >
        <h1
          className="typo-undertittel"
        >
          <span
            tabIndex={-1}
          >
            AREMARK TESTFAMILIEN
             (
            41
            )
          </span>
        </h1>
        <span
          className="visually-hidden"
        >
          Mann
        </span>
        <span
          className="typo-normal"
        >
          <span
            className="c5"
          >
            <span
              title="Fødselsnummer"
            >
              <span
                data-test-id="person-fnr"
              >
                10108000398
              </span>
              <span>
                <button
                  className="c6"
                  onClick={[Function]}
                >
                  <svg
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M15.5 23.5h-14v-20h9l5 5zm-5-20v5h5m-8-5v-3h9l5 5v15h-6m1-20v5h5"
                      fill="none"
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit="10"
                    />
                  </svg>
                  <span
                    className="sr-only"
                  >
                    Kopier fødselsnummer
                  </span>
                </button>
                <textarea
                  aria-hidden={true}
                  className="c7"
                  readOnly={true}
                  tabIndex={-1}
                  value="10108000398"
                />
              </span>
            </span>
          </span>
          <ul
            className="c8"
          >
            <li
              title="Statsborgerskap"
            >
              Norge
            </li>
            <li
              title="Sivilstand"
            >
              Gift
            </li>
            <span
              className="c9"
              title="Barn under 22 år"
            >
              <svg
                className="c10"
                title="Ett eller flere av barna har status som død"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fillRule="evenodd"
                  stroke="none"
                  strokeWidth={1}
                >
                  <g
                    transform="translate(-16.000000, -20.000000)"
                  >
                    <g
                      transform="translate(16.000000, 20.000000)"
                    >
                      <path
                        d="M12.2047909,-0.004 L11.990873,-0.00191299528 C8.77688569,0.054436132 5.75177235,1.35359657 3.47380671,3.65660627 C1.17914502,5.97631201 -0.0532312621,9.01290387 0.00207436285,12.2081081 C0.116859622,18.8207825 5.29784883,24.0007282 11.7967815,24.0007282 L12.0086125,23.9986412 C18.7350287,23.8828125 24.1142833,18.4044251 23.9984546,11.7896636 C23.8836693,5.17698921 18.7026801,-0.004 12.2047909,-0.004 Z"
                        fill="#FFA733"
                        fillRule="nonzero"
                      />
                      <path
                        d="M12.0271658,18.9998332 C12.0181663,18.9998332 12.0091668,18.9998332 12.0001673,18.9998332 C11.1827127,18.9998332 10.5137499,18.3458695 10.5002507,17.5269151 C10.4852515,16.6989611 11.1452148,16.0149992 11.9731688,16.0014999 C11.9821683,16.0014999 11.9911678,16 12.0001673,16 C12.8176218,16 13.4850847,16.6554636 13.5000839,17.474418 C13.515083,18.302372 12.8536198,18.984834 12.0271658,18.9998332 Z"
                        fill="#3E3832"
                        fillRule="evenodd"
                      />
                      <path
                        d="M12,5 C12.5522847,5 13,5.44771525 13,6 L13,13 C13,13.5522847 12.5522847,14 12,14 C11.4477153,14 11,13.5522847 11,13 L11,6 C11,5.44771525 11.4477153,5 12,5 Z"
                        fill="#3E3832"
                        fillRule="nonzero"
                      />
                    </g>
                  </g>
                </g>
              </svg>
              2
               barn under 22 år
            </span>
          </ul>
        </span>
      </section>
    </section>
    <div
      className="c11"
    >
      <section
        aria-label="etiketter"
        className="c12"
        role="region"
      >
        <div
          className="etikett etikett--advarsel"
        >
          <span
            className="typo-normal"
          >
            Sperret adresse, strengt fortrolig
          </span>
        </div>
        <div
          className="etikett etikett--advarsel"
        >
          <span
            className="typo-normal"
          >
            Egen ansatt
          </span>
        </div>
        <div
          className="etikett etikett--advarsel"
        >
          <span
            className="typo-normal"
          >
            Sikkerhetstiltak
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Reservert i KRR
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Manuell oppfølging
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Vergemål
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Talespråktolk
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Tegnspråktolk
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Dødsbo
          </span>
        </div>
        <div
          className="etikett etikett--fokus"
        >
          <span
            className="typo-normal"
          >
            Fullmakt
          </span>
        </div>
      </section>
      <section
        aria-label="Nav kontor"
        className="c13"
      >
        <h2
          className="typo-element"
        >
          NAV-kontor / 0219 NAV Bærum
        </h2>
      </section>
    </div>
  </div>
  <div
    className="c14"
  >
    <button
      aria-expanded={false}
      aria-label="Åpne visittkort (Alt + N)"
      className="c15"
      onClick={[Function]}
      title="Åpne visittkort (Alt + N)"
    >
      <span
        className="nav-frontend-chevron chevronboks chevron--ned"
      />
      <span
        className="visually-hidden"
      >
        Ekspander visittkort
      </span>
    </button>
  </div>
</section>
`;
